### DISCUSSION TITLE
Overlay support

### DISCUSSION THREAD

**Discussion by avidseeker** [ImageMagick/ImageMagick#7738] [Help]
### ImageMagick version

7.1.1

### Operating system, version and so on

GNU/Linux

### Description

Does imagemagick support any image format with the new overlay (multi-layer) feature?

- https://graphicdesign.stackexchange.com/questions/166219/
- https://graphicdesign.stackexchange.com/questions/166214/

**Comment by snibgo** [comment]
IM can write multiple images to single files that have extensions JXL, AVIF, WEBP, TIFF, GIF and some others.

**Reply by avidseeker** [reply]
Can you elaborate? I don't see this option in the `-layer` flag (https://imagemagick.org/script/command-line-options.php).

To clarify, my goal is:
1. merge multiple images as multiple layers in one file
2. view the nth frame of a file

**Comment by snibgo** [comment]
Suppose we want our output files to contain two versions of my toes image.
```
magick toes.png ( +clone -evaluate Multiply 0.7 ) out.jxl

magick toes.png ( +clone -evaluate Multiply 0.7 ) out.avif
```
We can verify that the output files contain two images:
```
magick out.jxl info:
out.jxl[0] JXL 267x233 267x233+0+0 16-bit sRGB 0.000u 0:00.000
out.jxl[1] JXL 267x233 267x233+0+0 16-bit sRGB 18395B 0.001u 0:00.001
```
IM can read just the Nth image from a file by using `[N]`, counting from zero.For example:
```
magick out.jpg[1] x.png
```

My version of Gimp (v2.10.38) can see only the first image in the JXL. It can see both images in the AVIF, but asks which one I want to open, rather than opening both as layers. Perhaps I misunderstand Gimp, or a more recent version can open all the images in an AVIF as layers (as it can for TIFF, for example).
